files/nginx.conf

```diff
@@ -0,0 +1,85 @@
+user www-data;
+worker_processes auto;
+pid /run/nginx.pid;
+include /etc/nginx/modules-enabled/*.conf;
+
+events {
+	worker_connections 768;
+	# multi_accept on;
+}
+
+http {
+
+	##
+	# Basic Settings
+	##
+
+	sendfile on;
```

Comment:
When we commit config files like this, it's hard to understand what all the settings do. I realise that you probably don't know what the settings do either, we just get this stuff from blogs right!? I do it all the time :D what I want to do for this project though is make sure we at least understand it a little. The best way to do this is to put comments next to each entry, e.g. I looked up what sendfile does

* https://docs.nginx.com/nginx/admin-guide/web-server/serving-static-content/
* https://thoughts.t37.net/nginx-optimization-understanding-sendfile-tcp-nodelay-and-tcp-nopush-c55cdd276765?gi=e6b6da5d6162

This does seem like something we want. The second page has this: "If you’re serving locally stored static files, sendfile is totally essential to speed your Web server". To understand what it does, I looked into it a little more, and it basically lets us copy files without this buffering step in between, which makes it faster. But I'll never remember that, and next time I look at this I will definitely ask myself "what does sendfile do?" :D 

So the answer is we put a comment. I would add something to the end of this line

```
sendfile on;                            # speeds up file downloads by streaming them directly instead of buffering [1]
```

So we have a short description, and then also that number. The number is a reference. Somewhere else at the bottom of the file, we are going to add more comments...

```
References
-----------

[1]  Enabling the sendfile directive eliminates the step of copying the data into the buffer
       and enables direct copying data from one file descriptor to another. This saves a bunch
       of resources & runs the command in the kernel, so it's faster. "If you’re serving locally
       stored static files, sendfile is totally essential to speed your Web server".

      * https://docs.nginx.com/nginx/admin-guide/web-server/serving-static-content/#enabling-sendfile
      * https://thoughts.t37.net/nginx-optimization-understanding-sendfile-tcp-nodelay-and-tcp-nopush-c55cdd276765?gi=e6b6da5d6162
 ```

This way instead of a file full of basically meaningless configuration, we can look at this file and it actually makes sense to us.

We may not finish all of them in this PR, but as coders this is the standard we ought to be trying to get to.

See if you can add some comments (and merge my one in) to a few of the other configurations. If you find one you don't understand, ask me about it, and we'll figure it out together